### PR TITLE
Allow to cache HEAD requests

### PR DESCRIPTION
**Pull request by thoean**

### CODE REVIEW

File: test/Cimpress.Extensions.Http.Caching.InMemory.UnitTests/InMemoryCacheFallbackHandler_when_sending_request.cs [hunk 1]
```diff
@@ -42,12 +42,29 @@ public class InMemoryCacheFallbackHandler_when_sending_request
...
         }
 
         [Fact]
+        public async Task Updates_the_cache_for_head_and_get_independently()
+        {
+            // setup
+            var testMessageHandler = new TestMessageHandler();
+            var cache = new Mock<IMemoryCache>(MockBehavior.Strict);
+            var cacheTime = TimeSpan.FromSeconds(123);
+            cache.Setup(c => c.CreateEntry(url));
```
Comment (wparad) [inline test/Cimpress.Extensions.Http.Caching.InMemory.UnitTests/InMemoryCacheFallbackHandler_when_sending_request.cs]: How are we setting up something different then we are verifying?

File: src/Cimpress.Extensions.Http.Caching.Redis/RedisCacheHandler.cs [hunk 2]
```diff
@@ -50,7 +50,7 @@ public class RedisCacheHandler : DelegatingHandler
         /// <returns>The HttpResponseMessage from cache, or a newly invoked one.</returns>
         protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
         {
-            var key = request.RequestUri.ToString();
+            var key = request.Method + request.RequestUri.ToString();
```
Comment (wparad) [inline src/Cimpress.Extensions.Http.Caching.Redis/RedisCacheHandler.cs:53]: Should we add a token between the method and the request url?

**Review by wparad — CHANGES_REQUESTED** [review]

File: src/Cimpress.Extensions.Http.Caching.Redis/RedisCacheHandler.cs (hunk 2 above)
Comment (thoean) [reply src/Cimpress.Extensions.Http.Caching.Redis/RedisCacheHandler.cs:53]: Pure cosmetics. I'd skip that.

File: test/Cimpress.Extensions.Http.Caching.InMemory.UnitTests/InMemoryCacheFallbackHandler_when_sending_request.cs (hunk 1 above)
Comment (thoean) [reply test/Cimpress.Extensions.Http.Caching.InMemory.UnitTests/InMemoryCacheFallbackHandler_when_sending_request.cs]: No, but I wanted to adjust the headline of the test. Yes, I could go through a couple more send options like `PATCH` or `OPTIONS` but I opted to skip that.

**Review by wparad — APPROVED** [review]

**Pull request merged**